### PR TITLE
Creating Index patterns via role

### PR DESCRIPTION
**Pull request by ssi444**
### Description
Added the ability to automatically create index patterns.

### Issues Resolved
https://github.com/opensearch-project/ansible-playbook/issues/66

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ssi444** [comment]
Calling @saravanan30erd @prudhvigodithi for the review.
Thanks.

**Comment by ssi444** [comment]
@peterzhuamazon @saravanan30erd @prudhvigodithi ping

**Comment by peterzhuamazon** [comment]
@saravanan30erd Please let me know if you have bandwidth to review this.
@ssi444 Sorry we are focusing on two minor releases in the past weeks that is why this is pretty slow.
Thanks.

**Comment by ssi444** [comment]
@peterzhuamazon @saravanan30erd @prudhvigodithi ping

File: README.md [hunk 1]
```diff
@@ -87,13 +91,73 @@ cluster_type: single-node
...
+### Custom configuration files
+
+To override the default settings files, you need to put your settings in the `files` directory. The files should be 
+named exactly the same as the original ones (internal_users.yml, roles.yml, tenants.yml, etc.)
+
+Especially note the file `files/internal_users.yml`. If it exists and the `copy_custom_security_configs: true` setting is enabled, 
+then only in this case the task of setting passwords for internal users from variables is started. If the file `internal_users.yml` 
+is not located in the `files` directory, but, for example, in one of its subdirectories, then playbook will not work correctly
+
+### IaC (Infrastructure-as-Code)
```
Comment (prudhvigodithi) [inline README.md:118]: Is this same as this PR @ssi444 ?
https://github.com/opensearch-project/ansible-playbook/pull/63/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5?

File: roles/linux/opensearch/templates/tlsconfig.yml [hunk 2]
```diff
@@ -2,13 +2,13 @@ ca:
...
       keysize: 2048
-      validityDays: 730
+      validityDays: {{ cert_valid_days }}
       pkPassword: none
       file: root-ca.pem
 
 ### Default values and global settings
 defaults:
-      validityDays: 730
+      validityDays: {{ cert_valid_days }}
```
Comment (prudhvigodithi) [inline roles/linux/opensearch/templates/tlsconfig.yml:11]: Same as this existing PR @ssi444 
https://github.com/opensearch-project/ansible-playbook/pull/63/files
Should we revisit this until the other is merged ?
@saravanan30erd @peterzhuamazon

Comment (ssi444) [reply roles/linux/opensearch/templates/tlsconfig.yml:11]: The code is divided into logical blocks at @saravanan30erd request, so there are 3 PR's. Practically, you can fill in the last one. @prudhvigodithi

**Comment by prudhvigodithi** [comment]
Hey @ssi444 can you elaborate this functionality, we expect users to add multiple indexes under a pattern, in this scenario how does this functionality helps? Also with this can we load an index to created pattern?

**Comment by ssi444** [comment]
> Hey @ssi444 can you please elaborate this functionality, we expect users to add multiple indexes under a pattern, in this scenario how does this functionality helps? Also with this can we load an index to created pattern?

@prudhvigodithi 
After sending logs is configured, indexes for incoming data are created in OpenSearch. But it is impossible to search these indexes until an IndexPattern is created. This change allows you to create the necessary IndexPattern in different Tenants based on the configuration file, and not manually. This allows this role to take another step to manage the OpenSearch cluster using the IaC methodology

**Comment by ssi444** [comment]
> Hey @ssi444 apologies for the delay, do you think we still need this change to move forward?

yes, I think this is the necessary functionality

**Comment by dblock** [comment]
Looks like a good feature, however the PR got stalled and wasn't merged, sorry about that. Closing it for now, if you feel it does need to be merged, please rebase, and I can try to assign it to a maintainer in this repo.

[Catch All Triage - Attendees [1](https://github.com/dblock/), [2](https://github.com/krisfreedain), [3](https://github.com/andrross), [4](https://github.com/gkamat), [5](https://github.com/getsaurabh02)]

**Pull request closed without merging**